pombola/south_africa/views.py

```diff
@@ -162,31 +163,14 @@ def get_object(self):
         return province
 
     def get_ward_councillors(self, location):
-        # Look up the ward on MapIt:
-        url = 'http://mapit.code4sa.org/point/4326/{lon},{lat}?type=WD'.format(
-            lon=location.x, lat=location.y
-        )
-        try:
-            r = requests.get(url, timeout=API_REQUESTS_TIMEOUT)
-        except requests.exceptions.RequestException as e:
-            raise WardCouncillorAPIDown(u"MapIt request failed: {0}".format(e))
-        mapit_json = r.json()
-        if not mapit_json:
+        # Look up the ward in our local MapIt mirror of the Code4SA
+        # ward data:
+        area = mapit.models.Area.objects.by_location(location) \
+            .filter(type__code='WD').first()
+        if not area:
             return []
-        ward_id = mapit_json.values()[0]['name']
-        # Then find the ward councillor from that ward ID. There
-        # should only be one at the moment, but make it a list in case
-        # we support broader lookups in the future:
-        url_fmt = 'http://nearby.code4sa.org/councillor/ward-{ward_id}.json'
-        try:
-            r = requests.get(
-                url_fmt.format(ward_id=ward_id),
-                timeout=API_REQUESTS_TIMEOUT)
-            r.raise_for_status()
-            ward_result = r.json()
-        except (requests.exceptions.RequestException, ValueError) as e:
-            raise WardCouncillorAPIDown(unicode(e))
-        councillor_data = ward_result['councillor']
```

Comment:
Although it looks like `ward_result` is used further down (line 190) so possibly this still needs to be the two step process.